tests/test_dialects.py

```diff
@@ -991,6 +1217,77 @@ def test_hive(self):
         self.validate(
             "CAST(1 AS INT)", "CAST(1 AS INT)", read="hive", write="starrocks"
         )
+        self.validate(
+            "MIXED_TYPE_TO_DATE_STR(x)",
+            "DATE_FORMAT(TO_DATE(SUBSTR(REPLACE(CAST(x as string), '-', ''), 1, 8), 'yyyyMMdd'), 'yyyy-MM-dd')",
+            write="hive",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_TO_DATE(x)",
+            "TO_DATE(SUBSTR(REPLACE(CAST(x as string), '-', ''), 1, 8), 'yyyyMMdd')",
+            write="hive",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_TO_DI(x)",
```

Comment:
should it be CAST(TO_DATE(x) AS INT)

Reply:
You can't cast a date to an int. So you first need to cast it to a string in dateint format and then to an int.